MembraneRTC/src/main/java/org/membraneframework/rtc/media/SoundDetection.kt

```diff
@@ -0,0 +1,201 @@
+package org.membraneframework.rtc.media
+
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import timber.log.Timber
+import java.util.*
+import kotlin.math.abs
+import kotlin.math.log10
+
+/**
+ * Class responsible for sound detection using the device's microphone.
+ */
+class SoundDetection {
+    private var timer: Timer? = null
+    private var audioRecord: AudioRecord? = null
+    private var onSoundDetectedListener: OnSoundDetectedListener? = null
+    private var bufferSize = 0
+    var isRecording = false
+        private set
+
+    /**
+     * Starts the sound detection process with the specified monitor interval, sampling rate, and volume threshold.
+     *
+     * @param monitorInterval The time interval (in milliseconds) between sound detection checks.
+     * @param samplingRate The audio sampling rate in Hz.
+     * @param volumeThreshold The threshold value in decibels (dB) above which a sound is considered detected.
+     */
+    fun start(monitorInterval: Int = 1, samplingRate: Int = 22050, volumeThreshold: Int = -60) {
```

Comment:
is `monitorInterval` by default = 1ms ok? 
Even if we wanted an update on every frame (which is a lot!) that would be 16ms.

Reply:
good point 👌 I increased the value to 50 ms